layers/state_tracker/cmd_buffer_state.h

```
@@ -406,9 +406,10 @@ class CMD_BUFFER_STATE : public REFCOUNTED_NODE {
         AddChild(base);
     }
     template <typename StateObject>
-    void AddChildren(vvl::span<std::shared_ptr<StateObject>> &child_nodes) {
+    void AddChildren(vvl::span<StateObject *> &child_nodes) {
```

Would it be a good approach to make `AddChildren` a helper function private to cpp file? Now that we use raw pointer instead of shared_ptr I guess we can have larger set of potential failure cases when someone decides to use it for other object types than buffers (the signature suggest it's general function for different object types). As I understand for buffers we have guarantee that pointers are safe due to DestroyBuffer handling, but it might not be the case for other object type.

---

I went for Jeremy's suggestion, indeed no need for a template here